python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -1010,19 +1010,40 @@ def resolve_latestMaterializationByPartition(
 
         partitions = partitions or self.get_partition_keys()
 
-        events_for_partitions = get_asset_materializations(
-            graphene_info,
-            self._external_asset_node.asset_key,
-            partitions,
-        )
+        if partitions is None or len(partitions) > 1000:
```

Comment:
will partitions ever be None on line 1013 given what happens on line 1011?

Relatedly - if partitions is None, do we still need to pass in the computed partitions on line 1011 to get_asset_materializations along with the storage IDs? Or does that make the underlying sql query more complicated?

Reply:
whoops partitions will never be None, you're right -- it's still nice to pass in the partitions even if the user is requesting all partitions because it's possible that there are partitions in storage which don't exist in the definition (e.g. you changed partitions defs at some point)

updated the code here